src/Cameras/VR/babylon.vrExperienceHelper.ts

```diff
@@ -1407,6 +1407,16 @@ module BABYLON {
                 // The object selected is the floor, we're in a teleportation scenario
                 if (this._teleportationInitialized && this._isTeleportationFloor(hit.pickedMesh) && hit.pickedPoint) {
                     // Moving the teleportation area to this targetted point
+
+                    //Raise Mesh unselected event if we trigger the floor meshes and a non-floor mesh was previously selected
+                    if (this._currentMeshSelected &&
+                        !this._isTeleportationFloor(this._currentMeshSelected) &&
+                        this.onSelectedMeshUnselected.hasObservers()) {

```

Comment:
No need to test for observers. It is done internally

Reply:
Fixed :P. Thanks for the clarification.